From 5d339781f71a37083d4c4d304ba45b89aede184f Mon Sep 17 00:00:00 2001
From: Rolf Neugebauer <rolf.neugebauer@docker.com>
Date: Wed, 11 Jan 2017 22:40:38 +0000
Subject: [PATCH 5/7] Drivers: hv: utils: Force TimeSync version 3.0 on Windows
 10

Some older Windows 10 builds, including 10586 do not seem to
support TimeSync protocol 4.0 causing loss of time synchronisation
and a lot of log spam.

Signed-off-by: Rolf Neugebauer <rolf.neugebauer@docker.com>
---
 drivers/hv/hv_util.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/hv/hv_util.c b/drivers/hv/hv_util.c
index e7707747f56d..5961f1a49deb 100644
--- a/drivers/hv/hv_util.c
+++ b/drivers/hv/hv_util.c
@@ -401,7 +401,12 @@ static int util_probe(struct hv_device *dev,
 	default:
 		util_fw_version = UTIL_FW_VERSION;
 		sd_srv_version = SD_VERSION;
-		ts_srv_version = TS_VERSION;
+		/*
+		 * XXX rneugeba: force TimeSync Version to 3.0 on
+		 * Windows 10 as well as older Windows 10 builds,
+		 * including 10586 to not support version 4.0
+		 */
+		ts_srv_version = TS_VERSION_3;
 		hb_srv_version = HB_VERSION;
 	}
 
-- 
2.11.0

